webhook/plan_webhook.go

```diff
@@ -0,0 +1,75 @@
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	v1 "k8s.io/api/admission/v1"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func ValidateWebhook(c client.Client) *webhook.Admission {
+	return &webhook.Admission{
+		Handler: admission.HandlerFunc(func(ctx context.Context, req webhook.AdmissionRequest) webhook.AdmissionResponse {
+			if req.Operation == v1.Create || req.Operation == v1.Update {
+				if len(req.Object.Raw) == 0 {
+					return webhook.Denied("Request object is empty")
+				}
+
+				plan, err := rawToPlan(req.Object)
+				if plan == nil || err != nil {
+					return webhook.Denied("Failed to parse request object into Plan: " + err.Error())
+				}
+
+				targetNamespace := plan.Spec.TargetNamespace
+
+				for _, verb := range []string{"get", "list", "create"} {
+					sar := &authorizationv1.SubjectAccessReview{
```

Comment:
The eventual plan is to replace this with the AggregatedAPIServer that returns either cluster or cluster/namespace for a role type?